packages/trpc/src/context.ts

```diff
@@ -1,22 +1,10 @@
 import * as trpcNext from "@trpc/server/adapters/next";
 import { SALEOR_AUTHORIZATION_BEARER_HEADER, SALEOR_API_URL_HEADER } from "@saleor/app-sdk/const";
 import { inferAsyncReturnType } from "@trpc/server";
-
-const getBaseUrl = (headers: { [name: string]: string | string[] | undefined }): string => {
-  const { host, "x-forwarded-proto": xForwardedProto = "http" } = headers;
-
-  const xForwardedProtos = Array.isArray(xForwardedProto)
-    ? xForwardedProto.join(",")
-    : xForwardedProto;
-  const protocols = xForwardedProtos.split(",");
-  // prefer https over other protocols
-  const protocol = protocols.find((el) => el === "https") || protocols[0];
-
-  return `${protocol}://${host}`;
-};
+import { getAppBaseUrl } from "@saleor/apps-shared";
 
 export const createTrpcContext = async ({ res, req }: trpcNext.CreateNextContextOptions) => {
-  const baseUrl = getBaseUrl(req.headers);
+  const baseUrl = getAppBaseUrl(req.headers);
```

Comment:
Would remove the `appId` form here and delegate adding it to the context to middlewares. 
Also the ssr flag might be not needed anymore - don't think we call tRPC backend side in any of the apps.

Reply:
good point, trpc should handle context extending hopefully

Reply:
Okay I can't do that, there was a reason for it. 

So I need in context to "declare" it like `appId:string | undefined` 

otherwise typescript fails because it never knows that this field ever existed

Reply:
Regarding ssr, I will first migrate and check if it is used or not